File: api/messages_read.py
from pymongo import MongoClient
import os, json
from bson import ObjectId

MONGO_URL = os.environ["MONGO_URL"]
DB_NAME = os.environ.get("DB_NAME", "portfolio_db")

client = MongoClient(MONGO_URL)
db = client[DB_NAME]
messages = db["messages"]

def handler(request, response):
    if request.method == "PUT":
        try:
            body = request.body.read().decode("utf-8")
            data = json.loads(body)
            msg_id = data.get("id")
            messages.update_one({"_id": ObjectId(msg_id)}, {"$set": {"is_read": True}})
            return response.json({"success": True})
        except Exception as e:
            return response.json({"error": str(e)}, status=500)
    return response.json({"error": "Method not allowed"}, status=405)

File: api/messages_read.js
import { MongoClient, ObjectId } from "mongodb";

const client = new MongoClient(process.env.MONGO_URL);
const dbName = process.env.DB_NAME || "portfolio_db";

export default async function handler(req, res) {
  if (req.method === "PUT") {
    try {
      const { id } = req.body;

      await client.connect();
      const db = client.db(dbName);
      const messages = db.collection("messages");

      await messages.updateOne(
        { _id: new ObjectId(id) },
        { $set: { is_read: true } }
      );

      res.status(200).json({ success: true });
    } catch (err) {
      res.status(500).json({ error: err.message });
    }
  } else {
    res.status(405).json({ error: "Method not allowed" });
  }
}
